packages/zoe/src/zoeService/zoe.js

```diff
@@ -239,6 +249,21 @@ function makeZoe(vatAdminSvc, zcfBundleName = undefined) {
               /** @type {Promise<AddSeatObj>} */ (addSeatObjPromiseKit.promise),
             ).addSeat(invitationHandle, zoeSeatAdmin, seatData, seatHandle);
           },
+          tellZCFToLinkSeat: (
+            invitationHandle,
+            zoeSeatAdmin,
+            proposal,
+            initialAllocation,
+          ) => {
+            return E(
+              /** @type {Promise<AddSeatObj>} */ (addSeatObjPromiseKit.promise),
```

Comment:
I realise you're copying the type coercion from above, but it would be better to remove both of them and instead annotate the original declaration as:

```js
      /** @type {PromiseRecord<AddSeatObj>} */
      const addSeatObjPromiseKit = makePromiseKit();
```

Reply:
> Agreed that this could be clearer, but the curlies around the original .then callback means that the call to updateFromNotifier is not awaited.

Oops. You are correct. This is an interesting misreading hazard I was not really aware of. I'll keep it in mind. Thanks.

Reply:
I copied markM's comment to where it belongs.

Done.